Bluetooth: Host: Clarify use of identity in docs

HaavardRei opened the pull request:
Clarifies the use of identity in `bluetooth.h` by specifying whether the documentation refers to the identity address, i. e. the actual address `bt_addr_le_t`, or the identity handle, i. e. an index corresponding to a specific identity address.

omkar3141 commented on include/zephyr/bluetooth/bluetooth.h:
```diff
@@ -443,71 +442,67 @@ void bt_id_get(bt_addr_le_t *addrs, size_t *count);
+ * created using it, remove any pairing keys or other data associated with it, and then
+ * create a new identity address in the same slot, based on the @a addr and @a irk parameters.
+ *
+ * @note The default identity address (corresponding to BT_ID_DEFAULT) cannot be reset, i.e. this
```
```suggestion
 * @note The default identity address (corresponding to BT_ID_DEFAULT) cannot be reset, and
```

HaavardRei replied:
Changes this and the case below to "and this" as "and API will return..." doesn't make sense.

omkar3141 commented on include/zephyr/bluetooth/bluetooth.h:
```diff
@@ -443,71 +442,67 @@ void bt_id_get(bt_addr_le_t *addrs, size_t *count);
+ *             to BT_ADDR_LE_ANY the stack will generate a new static random
+ *             address for the identity address and copy it to the given
  *             parameter upon return from this function (in case the
  *             parameter was non-NULL).
```
Text in paranthesis is not needed.
```suggestion
 *
```

omkar3141 commented on include/zephyr/bluetooth/bluetooth.h:
```diff
@@ -443,71 +442,67 @@ void bt_id_get(bt_addr_le_t *addrs, size_t *count);
+ * occupying, the bt_id_reset() API needs to be used.
  *
- * @note the default identity (BT_ID_DEFAULT) cannot be deleted, i.e. this
+ * @note The default identity address (corresponding to BT_ID_DEFAULT) cannot be deleted, i.e. this
```
```suggestion
 * @note The default identity address (corresponding to BT_ID_DEFAULT) cannot be deleted, and 
```

omkar3141 reviewed (changes requested):
Minor comments.

alwa-nordic commented:
Tangent info: There is a spec term "device identity" that appears defined like this:

> A device identity consists of the peer’s Identity Address and a local and peer’s IRK pair.

Essentially it's a row in the Resolve List of the Controller.

This PR does not use "device identity", and I don't think it applies either, so I'm sharing this just for info.

omkar3141 reviewed (approved):
Looks good to me.

HaavardRei committed:
Bluetooth: Host: Clarify use of identity in docs

Clarifies the use of identity in `bluetooth.h` by specifying whether the
documentation refers to the identity address, i. e. the actual address
`bt_addr_le_t`, or the identity handle, i. e. an index corresponding to
a specific identity address.

Signed-off-by: Håvard Reierstad <haavard.reierstad@nordicsemi.no>
